jscomp/syntax/src/res_core.ml

```diff
@@ -4276,6 +4276,21 @@ and parseEs6ArrowType ~attrs p =
             p.uncurried_config |> Res_uncurried.fromDotted ~dotted
           in
           let loc = mkLoc startPos endPos in
+          let arity =
+            (* Workaround for ~lbl: @as(json`false`) _, which changes the arity *)
+            match argLbl with
+            | Labelled _s ->
+              let typ_is_any =
+                match typ.ptyp_desc with
+                | Ptyp_any -> true
+                | _ -> false
+              in
+              let has_as =
+                Ext_list.exists typ.ptyp_attributes (fun (x, _) -> x.txt = "as")
+              in
+              if typ_is_any && has_as then arity - 1 else arity
```

Comment:
I think there can be multiple `@as() _`, right?

Reply:
This check is approximate: does not check that we're inside an external.
The same construction inside a normal type annotation which is not an external should have no effect (in that the rest of the compiler ignores it).

Reply:
It would be good to deprecate this whole construction in future.
Kind of hesitant to go for this workaround, unless it's really needed to unbreak something.

Reply:
Good stuff!

Reply:
Yeah we should get rid of this eventually. But for v11 we should have it imo.

Reply:
OK I've guarded it to only apply inside externals, so it won't accidentally break in the corner case where the annotations are used outside externals by mistake.